apps/site/assets/js/view-previous-events.js

```diff
@@ -1,17 +1,21 @@
 const setupViewPreviousEventsButton = function() {
-  const viewPreviousEventsButton = document.querySelector(
-    ".m-previous-events-button"
-  );
-  const hiddenEventList = [...document.querySelectorAll(".m-hidden-event")];
+  const viewPreviousEventsButtons = [
+    ...document.querySelectorAll("a.m-previous-events-button")
+  ];
+
+  viewPreviousEventsButtons.forEach(button => {
+    button.addEventListener("click", () => {
+      // show hidden events for this button
+      // because of how the list is laid out, we can just iterate through next <li> siblings until we find one without .m-hidden-event
+      let li = button.parentElement;
+      do {
+        li.classList.remove("m-hidden-event");
+        li = li.nextElementSibling;
+      } while (li && li.classList.contains("m-hidden-event") === true);
```

Comment:
Is doing it this way better because it eliminates one `document.querySelectorAll`?  

It seems more brittle than before, since this loop is so specific about where the button sits in relationship to the rest of the list.  On the other hand, this list format shouldn't change in the future, so maybe it doesn't hurt to assert this list format?  Does that question make sense?

Reply:
Not necessarily better, just different! We still need to match specific events to each button (before only the first button was revealing _all_ hidden events on the page). Your question/concern makes sense -- another idea I was considering was adding a custom attribute to the markup that we could use to match. Like a name on each button that would match an attribute we add to every event it's responsible for. I think I will update it to do it that way instead!

Reply:
Oh oh oh right, gosh.  My brain was still stuck on the "one button".  I totally understand why you looped the way you did, then!  But I'll keep an eye out for the next change

Reply:
@hannahpurcell Just push an update that will probably be less brittle!